internal/archive/archive.go

```diff
@@ -205,3 +204,95 @@ func WriteDirToTar(tw *tar.Writer, srcDir, tarDir string, uid, gid int, mode int
 		return nil
 	})
 }
+
+func WriteZipToTar(tw *tar.Writer, srcZip, basePath string, uid, gid int, mode int64) error {
```

Comment:
A lot of this code is duplicated from `WriteDirToTar` and can be merged with a little bit more work. For the sake of trying to get this feature in, we are deferring it for now.